src/cargo/ops/cargo_clean.rs

```diff
@@ -136,8 +136,24 @@ pub fn clean(ws: &Workspace<'_>, opts: &CleanOptions<'_>) -> CargoResult<()> {
     for pkg in packages {
         let pkg_dir = format!("{}-*", pkg.name());
 
-        // Clean fingerprints.
         for (_, layout) in &layouts_with_host {
+            // Clean entries of workspace members under target/doc.
+            for doc_member_entry in pkg.targets().iter().filter(|t| t.documented()) {
+                if doc_member_entry.name() == pkg.name().as_str() {
```

Comment:
I'm a bit confused as to why this is comparing a target with the package name.  A package name being equal to a target name isn't how to detect if something is a workspace member.  There is a `ws.is_member()` method for that.

Reply:
thank you for your reply.